main/src/main/scala/sbt/Defaults.scala

```diff
@@ -4403,6 +4403,39 @@ trait BuildExtra extends BuildCommon with DefExtra {
       sbtPluginExtra(dependency, sbtV, scalaV)
     }
 
+
+  /**
+   * Overrides a dependency as an sbt plugin for the specific sbt version sbtVersion and Scala version scalaVersion.
+   * Typically, use the default values for these versions instead of specifying them explicitly.
+   */
+  def overrideSbtPlugin(
+      dependency: ModuleID,
+      sbtVersion: String,
+      scalaVersion: String
+  ): Setting[Seq[ModuleID]] =
+    dependencyOverrides += sbtPluginExtra(dependency, sbtVersion, scalaVersion)
+
+  /**
+   * Overrides dependency as an sbt plugin for the specific sbt version sbtVersion.
+   * Typically, use the default value for this version instead of specifying it explicitly
+   */
+  def overrideSbtPlugin(dependency: ModuleID, sbtVersion: String): Setting[Seq[ModuleID]] =
+    dependencyOverrides += {
+      val scalaV = (update / scalaBinaryVersion).value
+      sbtPluginExtra(dependency, sbtVersion, scalaV)
+    }
+
+  /**
+   * Overrides dependency as an sbt plugin for the sbt and Scala versions configured by
+   * sbtBinaryVersion and scalaBinaryVersion scoped to update.
+   */
+  def overrideSbtPlugin(dependency: ModuleID): Setting[Seq[ModuleID]] =
```

Comment:
Thanks for the contribution. Could you add a scripted test in https://github.com/sbt/sbt/tree/1.9.x/sbt-app/src/sbt-test/project or https://github.com/sbt/sbt/tree/1.9.x/sbt-app/src/sbt-test/plugins to demonstrate that it works please?

Reply:
Since this has to do with resolving dependencies, I will need to make reference to outside plugins. Is that okay? Which should I choose?
I could also attempt to publish to local ivy by using some sbt-tests just to publish, but that means that test needs to happen prior to the one where they're imported, or it needs to all be in subprojects of the test, but that seems odd.

Reply:
Yea. That's fine. That's why I listed https://github.com/sbt/sbt/tree/1.9.x/sbt-app/src/sbt-test/plugins since those tests reference existing plugins.

Reply:
It took me much longer to make the tests than the change, but I suppose that's unavoidable here.
Feel free to critique my work, I just went with the first way that I got it working